src/platforms/common/blackpill-f4/README.md

```diff
@@ -81,3 +82,57 @@ mon freq 900k
+
+4. Flash the [Black Magic Debug Probe compatible platform] firmware at offset to board which is running BMD bootloader (then it reboots and jumps to firmware):
+
+`$ dfu-util -d 1d50:6018 -S 375F334C3032 --alt 0 -s 0x08004000:leave -D build-f401cc/blackmagic_blackpill_f401cc_firmware.bin`
+
+5. Verify the board enumerates, jumps between bootloader and firmware (hold PA0 onboard pushbutton active-low on reset to stay in bootloader) by `dfu-util --detach` and `dfu-util -s :leave`, verify the firmware works with GDB and BMDA HL-remote and bmpflash SPI.
+
+### Variant with meson and ST MaskROM for bootloader
+1. Create a build configuration for specific platform (WeActStudio MiniSTM32F401CC) with specific options (disable BMD bootloader flags and offsets, use ST MaskROM)
+
+`$ meson setup build-f401cc_maskrom --cross-file=cross-file/blackpill-f401cc.ini -Dbmd_bootloader=false`
+
+2. Compile the firmware
+
+`$ ninja -C build-f401cc_maskrom/ blackmagic_blackpill_f401cc_firmware.bin`
+
+3. Flash the [Black Magic Debug Probe compatible platform] firmware to blank board in ST MaskROM (BOOT0) mode (then it reboots and jumps to firmware), this will take a while:
+
+`$ dfu-util -d 0483:df11 -S 375F334C3032 --alt 0 -s 0x08000000:leave -D build-f401cc_maskrom/blackmagic_blackpill_f401cc_firmware.bin`
+
+4. Verify the board enumerates, jumps between MaskROM and firmware (hold PA0 onboard pushbutton active-low on reset to make it jump to MaskROM, or hold BOOT0 to enter MaskROM directly) by `dfu-util --detach` and `dfu-util -s :leave`, verify the firmware works with GDB and BMDA HL-remote and bmpflash SPI.
+
+## More details
+
+* Makefiles is the historical build system, and it also supports both variants: pass `BMD_BOOTLOADER=1` makeflag to switch into two-binaries configuration (was `BMP_BOOTLOADER=1` for a year between PR1508 and PR1843).
+* ST MaskROM is the read-only System Memory bootloader which starts up per BOOT0-triggered SYSCFG mem-remap. It talks USART AN2606 so you can use stm32flash etc. When USBOTG_FS is plugged, it spins up HSE 25 MHz and measures its frequency relative to HSI (RC 8 MHz with worse tolerance) and sometimes can misdetect it as 24 or 26 MHz then configure a wrong input PLL divider, resulting in broken USB DFU. New 8 MHz HSE boards should not be affected by this, but also they are not yet supported (needs a different pll config).
+* BMD bootloader is the port of BMP bootloader with dfu_f1.c internal flash driver swapped to dfu_f4.c (and different serialno algorithm etc.) which a) has a fixed per-board PLL config, no autodetection; b) understands buttons, drives LED, does not touch other GPIOs, talks USB DfuSe, ~~has MS BOS WINUSB descriptors~~, uses same libopencm3 code so you can verify hardware config via a smaller binary; c) erases and writes to internal Flash ~2.4x faster than MaskROM; d) all of that in first 8-9 KiB of first page of 16 KiB (of F2/F4/F7 flash), just like on `native`/`stlink`/`swlink` etc.
```

Comment:
Accepted. Does it? Current `dfu_init()` only calls `usbd_init()`. It also has to call that https://github.com/blackmagic-debug/blackmagic/blob/488bd37ca00d6cb2578e96d88af9a9df9edbb636/src/platforms/common/usb.c#L59-L60 and have descriptor structs expanded.
The paragraph is too wordy and unstructured, I'll review it in ReText, even if it doesn't have a CommonMark mode.

Reply:
Indeed, the bootloader does not yet support this, which is why we're not asking you to remove the strike-through - the wording nitpick is that the BOS (Binary Object Storage) is not MS specific, and the descriptors are not WINUSB descriptors but rather MS OS Descriptors that define bindings for WINUSB to interfaces. It's perfectly possible to bind other drivers to interfaces instead of WINUSB if we wanted, among other things those descriptors have the power to do.

Reply:
Ironically, this is now a ST side benefit, they get to ship signed drivers for their "ST BOOTLOADER", meanwhile BMD bootloader requires manual drivers association on anything Windows.